lib/puppet/provider/cisco_ace/nxapi.rb

```diff
@@ -0,0 +1,222 @@
+#
+# The NXAPI provider for cisco_acl.
+#
+# July, 2015
+#
+# Copyright (c) 2015 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'cisco_node_utils' if Puppet.features.cisco_node_utils?
+begin
+  require 'puppet_x/cisco/autogen'
+rescue LoadError # seen on master, not on agent
+  # See longstanding Puppet issues #4248, #7316, #14073, #14149, etc. Ugh.
+  require File.expand_path(File.join(File.dirname(__FILE__), '..', '..', '..',
+                                     'puppet_x', 'cisco', 'autogen.rb'))
+end
+
+Puppet::Type.type(:cisco_ace).provide(:nxapi) do
+  desc 'The NXAPI provider.'
+
+  confine feature: :cisco_node_utils
+  defaultfor operatingsystem: :nexus
+
+  mk_resource_methods
+
+  # Property symbol array for method auto-generation.
+  #
+  # NOTE: For maintainability please keep this list in alphabetical order and
+  # one property per line.
+  ACL_NON_BOOL_PROPS = [
+    :action,
+    :proto,
+    :src_addr,
+    :src_port,
+    :dst_addr,
+    :dst_port,
+    :option_format,
+  ]
+
+  ACL_BOOL_PROPS = [
+  ]
```

Comment:
Hi Chris, some of my functions have acl_bool_props referenced.. it's more like a placeholder for future reference.